extension/api.js

```diff
@@ -96,131 +283,25 @@ this.search = class extends ExtensionAPI {
       },
     });
 
-    function finish(result = false) {
-      Services.prefs.setBoolPref(RUN_ONCE_PREF, result);
-    }
-
-    // If the selected default is not the configured default for this region/locale, the
-    // user has changed default and we should not re-prompt.
-    let defaultEngine = await Services.search.getDefault();
-    let configuredDefault = await Services.search.originalDefaultEngine;
-    if (defaultEngine.name !== configuredDefault.name) {
-      console.log("reset-default-search: The default engine was already selected by the user.");
-      finish();
-      return;
-    }
+    // Ensure everything is started if this is an APP startup.
+    await searchInitialized;
+    await ExtensionSettingsStore.initialize();
```

Comment:
Move this `ExtensionSettingsStore.initialize();` inside `tryDefaultReset`, to avoid initialization of `ExtensionSettingsStore` until it's actually needed.